src/parser.rs

```diff
@@ -2649,16 +2649,16 @@ impl<'a> Parser<'a> {
         };
 
         // Not Sure if Top should be cheked here as well. Trino doesn't support TOP.
-        let is_l_parent = if distinct {
+        let is_l_paren = if distinct {
             self.consume_token(&Token::LParen)
         } else {
             false
         };
 
         let projection = self.parse_comma_separated(Parser::parse_select_item)?;
 
-        if is_l_parent {
-            self.consume_token(&Token::RParen);
+        if is_l_paren && !self.consume_token(&Token::RParen) {
```

Comment:
clippy was complaining (rightly) that the return value from `consume_token` was not used